libmount/meson.build

```diff
@@ -131,20 +131,22 @@ libmount_test_src_override = {
   'debug': 'init',
 }
 
-foreach libmount_test: libmount_tests
-  test_name = 'test_mount_' + libmount_test
-  exe = executable(
-    test_name,
-    'src/' + libmount_test_src_override.get(libmount_test, libmount_test) + '.c',
-    include_directories : [dir_include, dir_libblkid],
-    link_with : [lib__mount, lib_common, lib_blkid_static],
-    dependencies : lib__mount_deps,
-    c_args : ['-DTEST_PROGRAM'],
-  )
-  # the test-setup expects the helpers in the toplevel build-directory
-  link = meson.project_build_root() / test_name
-  run_command('ln', '-srf', exe.full_path(), link,
-    check : true)
-endforeach
+if program_tests
+  foreach libmount_test: libmount_tests
+    test_name = 'test_mount_' + libmount_test
+    exe = executable(
+      test_name,
+      'src/' + libmount_test_src_override.get(libmount_test, libmount_test) + '.c',
+      include_directories : [dir_include, dir_libblkid],
+      link_with : [lib__mount, lib_common, lib_blkid_static],
+      dependencies : lib__mount_deps,
+      c_args : ['-DTEST_PROGRAM'],
```

Comment:
Why not use the same `build_by_default` logic that is used elsewhere in the patch?